### DISCUSSION TITLE
Openssl3 with Python

### DISCUSSION THREAD

**Discussion by gobbledy-gook** [open-quantum-safe/oqs-demos#307] [Q&A] [answered]
I was trying to build a docker file for integrating `ops-provider` with `liboqs-python`. Now there are a few considerations here.

1. python's ssl should use openssl3 and not openssl1.1.1
2. ops-provider and openssl3 setup is pre-requisite
3. there might more which I am not sure of yet.

The first point is troubling me. The way which I found to make python use openssl3 instead of openssl1.1.1 was to build python from source so that it can use openssl3 libraries and binaries. But while doing that I am facing issues. Following are commands I am running as a part of my docker file to build python from source.

```
# build OpenSSL3
WORKDIR /opt/openssl
RUN LDFLAGS="-Wl,-rpath -Wl,${INSTALLDIR}/lib64" ./config shared --prefix=${INSTALLDIR} && \
    make ${MAKE_DEFINES} && make install_sw install_ssldirs;

# Building Python using OPENSSL3
WORKDIR /usr/local
RUN wget https://www.python.org/ftp/python/3.9.7/Python-3.9.7.tgz
RUN tar -xzf Python-3.9.7.tgz
RUN cd Python-3.9.7 && ./configure --with-openssl=${INSTALLDIR} --with-openssl-rpath=${INSTALLDIR}/lib64 --enable-optimizations && \
make && make install
```

One of the log messages I found was this which was weird ->
```
988.4 Python requires an OpenSSL 1.0.2 or 1.1 compatible libssl with X509_VERIFY_PARAM_set1_host().
```

Any help would be greatly appreciated. Thanks in advance.

**Comment by SWilson4** [comment]
Hi @gobbledy-gook! I see you closed this discussion; did you sort everything out?

**Reply by gobbledy-gook** [reply, marked as answer]
Actually, I posted this in haste, should've been more patient. I realised that python3.12 supports openssl3, hence closed the discussion.
